File: AI.Core/OperatorsImplimentation/BasicOperators/Power.cs
﻿using System;
using AI.Core.ChannelsImplimentation;

namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    public class Power : Operator
    {
        public Power(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 2;
        }

        public override void Action()
        {
            var value = Convert.ToUInt64(Math.Pow(EnterContacts[0].Value, EnterContacts[1].Value));
            ExitContacts.SetValue(value);
        }
    }
}

File: NativeCore/Power.cpp
#include "Power.h"
#include "stdafx.h"

//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System;
using namespace AI::Core::ChannelsImplimentation;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				Power::Power(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(2);
				}

				void Power::Action()
				{
					auto value = static_cast<unsigned long long>(pow(EnterContacts[0]->Value, EnterContacts[1]->Value));
					ExitContacts->SetValue(value);
				}
			}
		}
	}
}
